api/keystore/service.go

```diff
@@ -301,6 +301,52 @@ func (ks *Keystore) ImportUser(r *http.Request, args *ImportUserArgs, reply *Imp
 	return nil
 }
 
+// DeleteUserArgs are arguments for passing into DeleteUser requests
+type DeleteUserArgs struct {
+	Username string `json:"username"`
+	Password string `json:"password"`
+}
+
+// DeleteUserReply is the response from calling DeleteUser
+type DeleteUserReply struct {
+	Success bool `json:"success"`
+}
+
+// DeleteUser deletes user with the provided username and password.
+func (ks *Keystore) DeleteUser(_ *http.Request, args *DeleteUserArgs, reply *DeleteUserReply) error {
+	ks.lock.Lock()
+	defer ks.lock.Unlock()
+
+	ks.log.Verbo("DeleteUser called with %s", args.Username)
+
+	if args.Username == "" {
+		return errEmptyUsername
+	}
+
+	// check if user exists and valid user.
+	usr, err := ks.getUser(args.Username)
+	switch {
+	case err != nil || usr == nil:
+		return fmt.Errorf("user doesn't exists: %s", args.Username)
+	case !usr.CheckPassword(args.Password):
+		return fmt.Errorf("incorrect password for user %q", args.Username)
+	}
+
+	// TODO: user is deleted from user db but returned an error due to some other problem.
+	// valid scenario ? discuss with team.
```

Comment:
The user should only be deleted if all the DB operations return no error. To ensure this, performing multiple deletes requires use of a Batch. But as it is right now, this fine.

Reply:
got you. thanks.